archive_etherpads.py

```diff
@@ -34,163 +45,173 @@ def confirm(prompt=None, resp=False):
             .replace("/", "_")
             .replace(".", "_")
-            .split("_")
         )
-        if x != ""
-    )
-    return url
+        # removes double underscores and leading/trailing underscores
+        url = "_".join(x for x in url.split("_") if x != "")
+        good_list.append(url)
+    return good_list
 
 
 def get_pad_content(url, destination):
-    """Read Pad content and write to HTML, after that return found links."""
+    """Read Pad content and write to HTML and txt, after that return found links."""
     if "/p/" not in url:
-        logger.info("IGNORED {} is not a valid pad url".format(url.encode("utf-8")))
+        logger.info(f"IGNORED {url} is not a valid pad url")
         return []
 
     try:
-        r = requests.get(url + "/export/html", headers=HEADERS)
+        response = requests.get(f"{url}/export/txt", headers=HEADERS, timeout=TIMEOUT)
+        response.encoding = "utf-8"
+        r = requests.get(f"{url}/export/html", headers=HEADERS, timeout=TIMEOUT)
         r.encoding = "utf-8"
+    except requests.Timeout as e:
+        logger.error(f"Timeout for url '{url}': {e}", stack_info=True)
+        return []
     except requests.ConnectionError as e:
-        logger.error(e)
+        logger.error(f"ConnectionError for url '{url}': {e}", stack_info=True)
         return []
 
     if r.status_code == 429:
         delay = int(r.headers["Retry-After"])
-        logger.warn(
-            "got status code 429 (Too Many Requests), waiting for {} seconds".format(
-                delay
-            )
+        logger.warning(
+            f"got status code 429 (Too Many Requests), waiting for {delay} seconds, then retrying"
         )
         time.sleep(delay)
+        logger.info(f"Retrying {url} now")
+        return get_pad_content(url, destination)  # retry after delay
     elif r.status_code != 200:
-        new_url = "{}/p/{}_seenotrettung".format(
-            "https://fachschaften.rwth-aachen.de/etherpad", url.split("/")[-1]
-        )
-        s = "{} -> {}".format(url, new_url)
-        logger.warn("got status code {}, consider moving {}".format(r.status_code, url))
-        if confirm("give Time to save?", resp=True):
-            print(s)
-            confirm("Done?")
-            return [new_url]
-        else:
-            return []
-    # TODO: Don't hard-code URLs
+        logger.warning(f"got status code {r.status_code}, ignoring {url}")
+        return []
+        #new_url = "{}/p/{}_seenotrettung".format(
+        #    "https://fachschaften.rwth-aachen.de/etherpad", url.split("/")[-1]
+        #)
+        #s = f"{url} -> {new_url}"
+        #logger.warning(f"got status code {r.status_code}, consider moving {url}")
+        #if False:  # confirm("give Time to save?", resp=True):
+        #    print(s)
+        #    confirm("Done?")
+        #    return [new_url]
+        #else:
+        #    return []
+        # TODO: Don't hard-code URLs
+        # TODO: What is the seenotrettung about?
```

Comment:
at the time we created this project, there were some issues with etherpad servers. If a pad broke, one could still obtain the content via some "hack" (history mode or similar) and copy-paste it into a new one (seenotrettung called that time, term was created/evolved during a plenum) which would be the go-to pad then.

Reply:
That makes sense. Can I remove that part from the code then?

Reply:
depends if we're now on a stable etherpad server or anticipate that issue in the future again